automation/e2e-functests.setup.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -xeu
+
+# This script prepares an ephemeral cluster with
+# latest changes CNAO and latest commit KubeVirt on top.
+# And clones the desired component repository
+#
+# This script also exports:
+# $KUBECONFIG
+#  kubeconfig full path so you can use`kubectl` binary or
+#  CNAO's `./cluster/kubectl.sh` script directly.
+#
+# $TMP_PROJECT_PATH
+# Latest changes CNAO temp directory
+#
+# $TMP_COMPONENT_PATH
+# The desired component temp directory
+#
+# Example:
+# export COMPONENT="KubeMacPool"
+# source automation/e2e-functest.setup.sh
+# cd ${TMP_COMPONENT_PATH}
+# KUBECONFIG=$KUBECONFIG make functests
+
+source hack/components/git-utils.sh
+source hack/components/yaml-utils.sh
+
+source automation/check-patch.setup.sh
+
+teardown() {
+    #make cluster-down
+    # rm -rf "$kubevirt_temp_dir"
+    # rm -rf "$component_temp_dir"
+    echo hi
+}
+
+trap teardown EXIT SIGINT SIGTERM SIGSTOP
+
+export COMPONENT=${COMPONENT:-kubemacpool}
+
+# Move to latest changes CNAO project temp dir
+cd ${TMP_PROJECT_PATH}
+
+# Spin up Kubernetes cluster
+export KUBEVIRT_PROVIDER='k8s-1.17'
+make cluster-down cluster-up
+
+# Export .kubeconfig full path, so it will be possible
+# to use 'kubectl' directly from the component directory path
+export KUBECONFIG=$(./cluster/kubeconfig.sh)
+
+#Deploy CNAO latest changes
+make cluster-operator-push
+make cluster-operator-install
+
+# Deploy KubeVirt latest stable release
+export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases | grep tag_name | grep -v -- - | sort -V | tail -1 | awk -F':' '{print $2}' | sed 's/,//' | xargs)
```

Comment:
last time i checked the site, the suggested method was
`export KUBEVIRT_VERSION=$(curl -s https://api.github.com/repos/kubevirt/kubevirt/releases|grep tag_name|sort -V | tail -1 | awk -F':' '{print $2}' | sed 's/,//' | xargs)`

is the one you use is the official suggested one ?

Reply:
Yes i took it from here https://kubevirt.io/quickstart_minikube/
the difference is `grep -v -- - `, eventually it will skip `rc` versions

Reply:
i think the official site has it as well ? (which is preferred to look on imo)
one time it changed the API and changed the suggested method